communication/src/TlsWriteQueue.h

```diff
@@ -28,19 +28,22 @@
 #include "TlsDiagnostics.h"
 
 namespace bft::communication::tls {
-
+struct Header {
+  uint32_t msg_size;
+  NodeNum endpoint_num;
+};
 // Any message attempted to be put on the queue that causes the total size of the queue to exceed
 // this value will be dropped. This is to prevent indefinite backups and useless stale messages.
 // The number is very large right now so as not to affect current setups. In the future we will
 // have better admission control.
 static constexpr size_t MAX_QUEUE_SIZE_IN_BYTES = 1024 * 1024 * 1024;  // 1 GB
-static constexpr size_t MSG_HEADER_SIZE = 4;
+static constexpr size_t MSG_HEADER_SIZE = sizeof(Header);
 
 struct OutgoingMsg {
   OutgoingMsg(std::vector<uint8_t>&& raw_msg)
       : msg(raw_msg.size() + MSG_HEADER_SIZE), send_time(std::chrono::steady_clock::now()) {
     uint32_t msg_size = htonl(static_cast<uint32_t>(raw_msg.size()));
-    std::memcpy(msg.data(), &msg_size, MSG_HEADER_SIZE);
+    std::memcpy(msg.data(), &msg_size, sizeof(msg_size));
```

Comment:
Only the header is expected to be copied here. i.e. MSG_HEADER_SIZE.

Reply:
We are copying msg_size into msg.data(). Previously MSG_HEADER_SIZE 4 bytes i.e. uint32_t. Now MSG_HEADER_SIZE is 8 bytes by inclusion of NodeNum endpoint_num. We are still copying the msg_size, so kept copy to sizeof(msg_size). In next 4 bytes we can copy endpoint_num.

Reply:
We don't copy msg_size into msg.data() here; we copy Header. The code should look like:
OutgoingMsg(std::vector<uint8_t>&& raw_msg, NodeNum endpointNum)
      : msg(raw_msg.size() + MSG_HEADER_SIZE), send_time(std::chrono::steady_clock::now()) {
      auto const msgSize = htonl(static_cast<uint32_t>(raw_msg.size() + MSG_HEADER_SIZE));
      auto const endpoint = htonl(endpointNum);
      const Header header{msgSize, endpoint};
      std::memcpy(msg.data(), &header, MSG_HEADER_SIZE);
      std::memcpy(msg.data() + MSG_HEADER_SIZE, raw_msg.data(), raw_msg.size());
  }

Reply:
endpointNum should be retrieved from the header in AsyncTlsConnection::readMsgSizeHeader and passed to the MsgReceiver

Reply:
Got it & updated. Thanks